File: app/src/main/java/ru/agpu/artikproject/background_work/settings_layout/user_background/UserBackgroundCheckBoxListener.java
package ru.agpu.artikproject.background_work.settings_layout.user_background;

import android.app.Activity;
import android.widget.CheckBox;

import ru.agpu.artikproject.background_work.datebase.MySharedPreferences;

public class UserBackgroundCheckBoxListener {
    /**
     * Слушатель нажатий на userBackgroundCheckBox
     *
     * @param act                    Активити
     * @param userBackgroundCheckBox Чекбокс
     */
    public UserBackgroundCheckBoxListener(Activity act, CheckBox userBackgroundCheckBox) {
        userBackgroundCheckBox.setOnCheckedChangeListener((buttonView, isChecked) -> {
            MySharedPreferences.put(act.getApplicationContext(), "enable_background_user", isChecked);
            new UserBackgroundCheckboxSetVisible(act, isChecked);
        });
    }
}

File: app/src/main/java/ru/agpu/artikproject/background_work/settings_layout/user_background/UserBackgroundCheckBoxListener.kt
package ru.agpu.artikproject.background_work.settings_layout.user_background

import android.app.Activity
import android.widget.CheckBox
import android.widget.CompoundButton
import ru.agpu.artikproject.background_work.datebase.MySharedPreferences

/**
 * Слушатель нажатий на userBackgroundCheckBox
 *
 * @param act                    Активити
 * @param userBackgroundCheckBox Чекбокс
 */
class UserBackgroundCheckBoxListener(act: Activity, userBackgroundCheckBox: CheckBox) {
    init {
        userBackgroundCheckBox.setOnCheckedChangeListener { _: CompoundButton?, isChecked: Boolean ->
            MySharedPreferences.put(act.applicationContext, "enable_background_user", isChecked)
            UserBackgroundCheckboxSetVisible(act, isChecked)
        }
    }
}